ExtractData/PKHeX.ExtractData/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace PKHeX.ExtractData
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string folder = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+            if (Path.GetFileName(folder).ToLower() == "debug")
+                folder = Directory.GetParent(folder).FullName;
+            if (Path.GetFileName(folder).ToLower() == "release")
+                folder = Directory.GetParent(folder).FullName;
+            if (Path.GetFileName(folder).ToLower() == "bin")
+                folder = Directory.GetParent(folder).FullName;
+            if (!Directory.Exists(folder)) return;
+            if (!Directory.Exists(folder + "\\Logs\\")) Directory.CreateDirectory(folder + "\\Logs\\");
+
+            ExtTMHM3 TMHM3 = new ExtTMHM3();
+            ExtLearnSet3 Learn3Data = new ExtLearnSet3();
+            ExtMoveTutors3 MoveTutors3 = new ExtMoveTutors3();
+            ExtEggMoves3 EggMoves3Data = new ExtEggMoves3();
+            
+            TMHM3.ExtractTMHM(folder + "\\Logs\\", folder + "\\GBA\\", folder + "\\Output\\");
```

Comment:
It's possible that the C# compiler will optimize this, but to improve performance, I would recommend assigning these "folder + ..." expressions to variables since you're referring to them several more times.